controller/commentsController.js

```diff
@@ -7,6 +7,62 @@ const CustomError = require("../utils/customError");
 // const User = require("../models/users");
 const responseHandler = require("../utils/responseHandler");
 
+exports.upvoteComment = async (req, res, next) => {
+  try {
+    const { commentId } = req.params;
+    const { ownerId } = req.body;
+
+    if (!mongoose.Types.ObjectId.isValid(commentId)) {
+      next(new CustomError(422, "invalid ID"));
+      return;
+    }
+    const comment = await Comments.findById({ _id: commentId });
+
+    //if user exists in downvotes array
+    if (comment.downVotes.includes(ownerId)) {
+      //get index of user in downvotes array
+      const voterIndex = comment.downVotes.indexOf(ownerId);
+      //if index exists
+      if (voterIndex > -1) {
+        //delete that index
+        comment.upVotes.splice(voterIndex, 1);
+      }
+    }
+
+    //same as above for upvotes
+    if (comment.upVotes.includes(ownerId)) {
+      const voterIdx = comment.upVotes.indexOf(ownerId);
+      if (voterIdx > -1) {
+        comment.upVotes.splice(voterIdx, 1);
+      }
+    }
+
+    // add user to the top of the upvotes array
+    comment.upVotes.unshift(ownerId);
```

Comment:
The `comment.upVotes.unshift(ownerId)` statement should be run **ONLY** when the ownerId was not in the list of upVotes already. Right now you remove the ownerId from the list onlly to add it back in a few lines later.

```suggestion
    //same as above for upvotes
    if (comment.upVotes.includes(ownerId)) {
      const voterIdx = comment.upVotes.indexOf(ownerId);
      if (voterIdx > -1) {
        comment.upVotes.splice(voterIdx, 1);
      }
    } else {
      // add user to the top of the upvotes array
      comment.upVotes.unshift(ownerId);
    }
```